utils/log/loggers/fields.go

```diff
@@ -46,8 +54,26 @@ func FromContext(ctx context.Context) LogFields {
 		return nil
 	}
 	if h := ctx.Value(contextKey); h != nil {
-		if logData, ok := h.(LogFields); ok {
-			return logData
+		if plf, ok := h.(*ProtectedLogFields); ok {
+			plf.mtx.RLock()
+			defer plf.mtx.RUnlock()
+			content := make(LogFields)
+			for k, v := range plf.Content {
+				content[k] = v
+			}
+			return content
+		}
+	}
```

Comment:
to avoid duplicated logic, would it be better to wrap this block using `fromContext`?

```golang
if plf := fromContext(ctx); plf != nil {
    plf.mtx.RLock()
    defer plf.mtx.RUnlock()
    content := make(LogFields)
    for k, v := range plf.Content {
        content[k] = v
    }
    return content
}
```

Reply:
Great idea, changed, thanks!